### PR TITLE
Fix Test Infrastructure

### PR DESCRIPTION
**Pull request by WernerMairl**
### What does this PR do?
Fixes problems with false positive tests

Changes: 
- call "AddApplicationPart()" for MVC => it helps MVC to find additional assemblies!
- a first test for Searchcontroller
- add some missing/broken config option!

### CODE REVIEW

File: src/BaGet/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -99,7 +99,7 @@ public static IServiceCollection AddBaGetContext(this IServiceCollection service
 
         public static IServiceCollection ConfigureHttpServices(this IServiceCollection services)
         {
-            services.AddMvc();
+            services.AddMvc().AddApplicationPart(typeof(BaGet.Web.Routes).Assembly);
```
Comment (loic-sharma) [inline src/BaGet/Extensions/ServiceCollectionExtensions.cs]: I haven't had any issues loading controllers from `BaGet.Web`. How did you run into these issues?

Also, I'm considering moving the controllers back to the `BaGet` assembly and renaming `BaGet.Web` to `BaGet.Protocol`. What do you think? The discussion for that is here: https://github.com/loic-sharma/BaGet/issues/73

Comment (loic-sharma) [reply src/BaGet/Extensions/ServiceCollectionExtensions.cs]: Could you add a comment like:

```
// TODO: Delete this
// See: https://github.com/loic-sharma/BaGet/issues/73
```

Comment (WernerMairl) [reply src/BaGet/Extensions/ServiceCollectionExtensions.cs]: comment added and added a few informations (links) to 73

Comment (tomzo) [reply src/BaGet/Extensions/ServiceCollectionExtensions.cs]: I just ran into this. Definitely needed.
@loic-sharma please merge this. It just adds a test and fixes assembly loading so no point in waiting.

**Comment by tomzo** [comment]
@WernerMairl maybe you could rebase on master.

**Comment by loic-sharma** [comment]
The controllers were brought back into the BaGet assembly, so this change is no longer needed. Thanks for opening this @WernerMairl!

**Pull request closed without merging**